### PR TITLE
Created a CONTRIBUTING document to outline project standards for cont…

### PR DESCRIPTION
**Pull request by shafferz**
# Description of Pull Request

This pull request proposes to add a `CONTRIBUTING.md` document to the `gatorgrouper/` root directory. This allows GitHub to automatically display a link to our contributing guidelines when making a pull request. This guide was contributed to by the leads of each team and their team members. 

You can [click here](https://help.github.com/en/articles/setting-guidelines-for-repository-contributors) to read more about creating a `CONTRIBUTING.md` document, its value for a software project, and examples of other similar guides.

Fixes #IssueNumber: None

## Type of Change

Please describe the pull request as one of the following: 

- [ ] Bug fix
- [ ] Breaking change
- [ ] New feature
- [x] Documentation update
- [ ] Other (Please specify any helpful information below)

## Tags

@gkapfham @Michionlion @toccinAC @sutterj @Alex-Yarkosky

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+should,
+  * Be written in a modular fashion, making use of functions and modules.
+  * Be clearly documented with both docstring comments describing functions, as
+    well as single line comments describing particularly complex lines of code.
+  * Pass `flake8` checks without errors. You can learn more about using `flake8`
+    [here](http://flake8.pycqa.org/en/latest/).
+  * Score 10.0/10.0 when using the `pylint` tool. You can learn more about using
+    `pylint` [here](https://www.pylint.org/).
+
+It is strongly encouraged that you use the Python code formatter,
```
Comment (Michionlion) [inline CONTRIBUTING.md]: Is it not required? `.travis.yml` fails anything that is not in the correct `black` format.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+The definition of complexity we use is taken from “A Philosophy of Software
+Design” by Chairman of Electric Cloud and Professor of Computer Science at
+Stanford, John Ousterhout. John defines complexity as anything “related to the
+structure of a software system that makes it hard to understand and modify the
+system.” To this end, the standards here aim to minimize complexity by
+maximizing the clarity and ease of modification of GatorGrouper. All
+contributions to GatorGrouper should have the following qualities:
+
+1. Changeability -
+*Software will be easily changeable to allow for future scalability.*
```
Comment (Michionlion) [inline CONTRIBUTING.md]: Should this use *`shall`*?

File: CONTRIBUTING.md (hunk 1 above)
Comment (shafferz) [reply CONTRIBUTING.md]: I was unsure of the relation between `black` and `pylint`, and had assumed that the only way to get a 10.0/10.0 on `pylint` was to use `black`. Is this incorrect? If so, I will make the use of the code formatter `black` more overtly necessary.

File: CONTRIBUTING.md (hunk 2 above)
Comment (shafferz) [reply CONTRIBUTING.md]: That, it should. Good catch! Fixed in a commit.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Michionlion) [reply CONTRIBUTING.md]: It is not strictly correct -- they are related, but do not require one another. I would mention both.

Comment (shafferz) [reply CONTRIBUTING.md]: Okay, I have disambiguated the black-pylint requirements in a new commit. @Michionlion review at your disclosure.

**Comment by Michionlion** [comment]
Since this PR does not change any code, I am merging it with 3 approvals.

**Pull request merged**